src/super_gradients/training/utils/callbacks.py

```diff
@@ -195,30 +197,79 @@ def __init__(self, email, model_meta_data, optimization_request_form, password=N
         password = password or getpass.getpass()
         self.platform_client.login(email, password)
 
+    @staticmethod
+    def log_optimization_failed():
+        logger.info(f"We couldn't finish your model optimization. Visit https://console.deci.ai for details")
+
+    def upload_model(self, model):
+        """
+            This function will upload the trained model to the Deci Lab
+
+            Args:
+                model: The resulting model from the training process
+        """
+        self.platform_client.add_model(
```

Comment:
For some architectures we must call prep_model_for_conversion().
Please check if we need to do so before passing the model through local_loaded_model arg by:
model = model.module.cpu()
if hasattr(model, "prep_model_for_conversion"):
    model.prep_model_for_conversion(input_size=self.model_met_data.input_dimensions)

(not sure this is how to get the input dimensions from ModelMetaData but you get the principle...)